test/Confluent.Kafka.IntegrationTests/Tests/AdminClient_NullReferenceChecks.cs

```diff
@@ -0,0 +1,123 @@
+            {
+                try
+                {
+                    adminClient.CreateTopicsAsync(new TopicSpecification[] { new TopicSpecification { Name = topicName1, NumPartitions = 1, ReplicationFactor = 1 } }).Wait();
+                    adminClient.CreatePartitionsAsync(new List<PartitionsSpecification> { new PartitionsSpecification { Topic = nullTopic, IncreaseTo = 2 } }).Wait();
+                    Assert.True(false, "Expected exception.");
+                }
+                catch (NullReferenceException ex)
+                {
+                    Assert.Contains("topic", ex.Message.ToLower());
+                }
+            }
+
+            // test adding a null list of brokers throws null reference exception.
+            using (var producer = new ProducerBuilder<Null, Null>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
+            using (var adminClient = new DependentAdminClientBuilder(producer.Handle).Build())
+            {
+                try
+                {
+                    adminClient.AddBrokers(null);
+                    Assert.True(false, "Expected exception.");
+                }
+                catch (NullReferenceException ex)
+                {
+                    Assert.Contains("broker", ex.Message.ToLower());
+                }
+            }
+
+            // test retrieving metadata for a null topic
+            using (var producer = new ProducerBuilder<Null, Null>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
+            using (var adminClient = new DependentAdminClientBuilder(producer.Handle).Build())
+            {
+                try
+                {
+                    adminClient.GetMetadata(null, TimeSpan.FromSeconds(10));
+                    Assert.True(false, "Expected exception.");
+                }
+                catch (ArgumentNullException ex)
+                {
+                    Assert.Contains("value cannot be null", ex.Message.ToLower());
+                }
+            }
+
+            // This test-case completing is the test case
+            // Used to cause a seg-fault.
+            using (var producer = new ProducerBuilder<Null, Null>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
+            using (var adminClient = new DependentAdminClientBuilder(producer.Handle).Build())
+            {
+                adminClient.DeleteTopicsAsync(new List<string> { topicName1, nullTopic });
+                Assert.True(true);
+            }
+
+            // Asserts existing ListGroup behavior - null group is valid option
+            using (var producer = new ProducerBuilder<Null, Null>(new ProducerConfig { BootstrapServers = bootstrapServers }).Build())
+            using (var adminClient = new DependentAdminClientBuilder(producer.Handle).Build())
+            {
+                adminClient.ListGroup(null, TimeSpan.FromSeconds(10));
```

Comment:
I think we should change this to throw an `ArgumentNullException`

Reply:
Implemented.